File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def deleteDuplicates(self, head):
        dummy=ListNode()
        tail=dummy
        while head:
            if head.next and (head.val==head.next.val):
                head.next=head.next.next
                continue
            tail.next=head
            head=head.next
            tail=tail.next
        return dummy.next
            
            
            
        

File: 83-remove-duplicates-from-sorted-list/remove-duplicates-from-sorted-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteDuplicates(ListNode head) {
       HashSet<Integer> set=new HashSet<>();
       ListNode prev=null;
       ListNode curr=head;
       while(curr!=null){
        if(set.contains(curr.val)){
            prev.next=curr.next;
        }
        else{
            set.add(curr.val);
            prev=curr;
        }
        curr=curr.next;
       }
       return head; 
    }
}